### ISSUE TITLE
Visual Studio incorrectly suggests "Open declaration can be removed"

### ISSUE THREAD

**Issue by nikonthethird** [Microsoft/visualfsharp#3306]
Visual Studio 2017 with the latest Updates shows "Open declaration can be removed" suggestion even when a type from the opened module is used.

#### Repro steps

Put the following code into an `.fsx` file and open it in Visual Studio. The line `open DeclaresA` will be grayed out.

```fsharp
module DeclaresA =
    type A = class end

module UsesA =
    open DeclaresA

    let a = Unchecked.defaultof<A>
```

![Image](http://i.imgur.com/OXKAprI.png)

#### Expected behavior

It should not be grayed out.

#### Actual behavior

It is grayed out, the tooling thinks nothing of the opened module is used.

#### Known workarounds

Since it is just a suggestion, simply ignoring it works.

#### Related information 

* Windows 10
* .NET 4.5
* Visual Studio 2017 with all Updates installed

**Comment by nikonthethird** [comment]
Something's also not quite right when using type aliases:

```fsharp
module DeclaresA =
    // type A () = class end // (1)
    type A = System.Int32 // (2)

open DeclaresA

let a = A ()
```

If the type `A` marked with (1) is uncommented, everything is fine.
If the type alias `A` marked with (2) is uncommented as shown, Visual Studio will once again suggest to remove the open statement.

**Comment by cartermp** [comment]
This appears fixed on all counts with 15.3.5. 

![fixed-1](https://user-images.githubusercontent.com/6309070/31304268-fa80fa1e-aad2-11e7-819a-aea12e9566ea.PNG)

![fixed-2](https://user-images.githubusercontent.com/6309070/31304269-fd669c66-aad2-11e7-8fb6-657dab191dae.PNG)

![fixed-3](https://user-images.githubusercontent.com/6309070/31304270-ff7cfbee-aad2-11e7-82bb-3b8b68465213.PNG)
